Sources/Kafka/KafkaConsumer.swift

```diff
@@ -331,26 +352,89 @@ public final class KafkaConsumer: Sendable, Service {
             try self.subscribe(topics: topics)
         }
 
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+                try await self.eventRunLoop()
+            }
+
+            group.addTask {
+                try await self.messageRunLoop()
+            }
+
+            // Throw when one of the two child task throws
+            try await group.next()
+            try await group.next()
```

Comment:
We probably want to do a `group.cancelAll()` after the first returned without throwing.

Reply:
I'm not sure, what about the following case:

When the client application stops reading the `KafkaConsumerMessages` sequence the `messageRunLoop()` will return. However, the `eventRunLoop()` might still be processing a _consumer close_ so I don't see a benefit in cancelling here, but open to discussion